lib/ruby_lsp/requests/base_request.rb

```diff
@@ -1,24 +1,26 @@
-# typed: true
+# typed: strict
 # frozen_string_literal: true
 
 module RubyLsp
   module Requests
     # :nodoc:
     class BaseRequest < SyntaxTree::Visitor
-      def self.run(document)
-        new(document).run
-      end
+      extend T::Sig
+      extend T::Helpers
+
+      abstract!
 
+      sig { params(document: Document).void }
       def initialize(document)
         @document = document
 
         super()
       end
 
-      def run
-        raise NotImplementedError, "#{self.class}#run must be implemented"
-      end
+      sig { abstract.returns(T.untyped) }
```

Comment:
I think `Object` would work better here:

```suggestion
      sig { abstract.returns(Object) }
```

It is still type safe and even safer since Sorbet will yell if you try to call non-existing methods `Object` rather than be silent because of the `T.untyped`.

Reply:
Some requests return `T.any`, which is incompatible with `Object`. Sorbet throws a runtime error. I think we'll need to keep this one as untyped.

Reply:
Got it to work in the latest commit.